### PR TITLE
Evaluation of energy dep temporal model

### PR DESCRIPTION
**Pull request by AtreyeeS**
Continues from #4272

### CODE REVIEW

File: gammapy/modeling/models/temporal.py [hunk 1]
```diff
@@ -657,6 +679,53 @@ def to_dict(self, full_output=False, format="table"):
...
+                )
+
+            m = RegionNDMap.create(region=None, axes=[time_axis, energy_axis])
+            kwargs.setdefault("marker", "None")
+            kwargs.setdefault("ls", "-")
+            m.quantity = self.evaluate(
+                time=time_axis.time_mid, energy=energy_axis.center
+            )
+            ax = m.plot(axis_name="time", ax=ax, **kwargs)
+            ax.set_ylabel("Norm / A.U.")
```
Comment (fabiopintore) [inline gammapy/modeling/models/temporal.py:725]: shouldn't this be in flux units if the model is energy dependent?

**Review by fabiopintore — COMMENTED** [review]
Thanks @AtreyeeS . I've left some inline comments.

File: gammapy/modeling/models/temporal.py (hunk 1 above)
Comment (AtreyeeS) [reply gammapy/modeling/models/temporal.py:725]: Well, we still take the unit from the spectral model, no?

File: gammapy/modeling/models/utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    with fits.open(filename) as hdul:
+        frame = hdul[0].header.get("frame", "icrs")
+        position = SkyCoord(
+            hdul[0].header["LONG"] * u.deg, hdul[0].header["LAT"] * u.deg, frame=frame
+        )
+
+        energy_hdu = hdul["ENERGIES"]
+        energy_axis = MapAxis.from_nodes(
+            nodes=energy_hdu.data,
+            unit=energy_hdu.header["TUNIT1"],
```
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply gammapy/modeling/models/utils.py:25]: We dont do this for the other templates, but ensure that `SkyModel.evaluate()` gives correct units

**Review by fabiopintore — APPROVED** [review]

File: gammapy/modeling/models/temporal.py [hunk 3]
```diff
@@ -657,6 +683,53 @@ def to_dict(self, full_output=False, format="table"):
...
 
+    def plot(self, time_range, energy=None, ax=None, n_points=100, **kwargs):
+        """
+        Plot Temporal Model.
+
+        Parameters
+        ----------
+        time_range : `~astropy.time.Time`
+            times to plot the model
+        energy : `~astropy.units.quantity`
```
Comment (registerrier) [inline gammapy/modeling/models/temporal.py:694]: I would move the `energy` argument to later to follow the arguments of `TemporalModel.plot`

**Review by registerrier — DISMISSED** [review]
Thanks @AtreyeeS . See some inline comments.

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . No more comments.

**Pull request merged**